### PR TITLE
Allow user to switch dark light mode

### LINKED ISSUE #160: allow user to switch dark/light mode
add a switcher to the user settings -> general to allow the user to switch between light and dark mode - and make ionic change accordingly. Similarly as the language-setting, set the dark/light mode when the user logged in/the system started.

### PR DESCRIPTION
**Pull request by ramandeepR**
fixes #160 .

### CODE REVIEW

File: src/templates/app.html [hunk 1]
```diff
@@ -26,7 +26,7 @@
     <script>
       window.AFFINITY_DEFAULT_TEAM = "<%= VUE_APP_DEFAULT_TEAM %>";
     </script>
-    <div id="app"></div>
+    <div id="app" class="dark"></div>
```
Comment (gnunicorn) [inline src/templates/app.html:29]:
```suggestion
    <div id="app"></div>
```

I don't think we need this.

File: src/views/Team/Outlet.vue [hunk 2]
```diff
@@ -24,6 +24,9 @@
                     <ion-icon size="small" :icon="editIcon" />
```
Comment (gnunicorn) [inline src/views/Team/Outlet.vue:24]: I think the changes in this file don't belong into this PR.

File: src/views/settings/General.vue [hunk 3]
```diff
@@ -32,11 +45,18 @@ import {
 import { defineComponent, computed } from 'vue';
 import { useStore } from '@/stores/';
 
+const SUPPORTED_THEMES = [
+  ["dark", "Dark Theme"],
+  ["light", "Light Theme"],
```
Comment (gnunicorn) [inline src/views/settings/General.vue:50]: this should also be translated.

**Review by gnunicorn — CHANGES_REQUESTED** [review]
Nice,

but if I understand [the docs correctly](https://ionicframework.com/docs/theming/dark-mode#manually-toggle-dark-mode), what is implemented here can only force dark-mode. Even the system is set to light mode, because the css is still enabled, if the device is configured to use it, the CSS would enable dark-mode. That isn't a problem per-se,  but the setting is confusing from a UX perspective..

How about rather than it being a configurable theme, we opt for a `forceDarkMode`-boolean and  have the setting use a ion-toggle with the same name? Then we can switch on dark-mode but not necessarily have to turn on light-mode.

File: src/templates/app.html (hunk 1 above)
Comment (ramandeepR) [reply src/templates/app.html]: Resolved

File: src/views/Team/Outlet.vue (hunk 2 above)
Comment (ramandeepR) [reply src/views/Team/Outlet.vue]: Resolved

File: src/config/App.vue [hunk 4]
```diff
@@ -92,11 +92,12 @@ export default defineComponent({
     if (store.state.auth.wantsToLogin) {
       updateLoginModal(true, false);
     }
-
+        
     watch(
       () => store.state.auth.user,
       async (newVal, oldVal) => {
         if (newVal && newVal != oldVal) {
+          document.body.classList.toggle('dark', newVal?.settings?.themes);
```
Comment (gnunicorn) [inline src/config/App.vue:100]: let's call the variable `force_darkmode`
```suggestion
          document.body.classList.toggle('dark', newVal?.settings?.force_darkmode);
```

File: src/locales/de.json [hunk 5]
```diff
@@ -171,6 +171,7 @@
   "richEditor.actions.unorderedListIcon" : "&#8226",
   "richtEditor.defaults.placeholder" : "Text...",
   "setting.general.language": "Sprache",
+  "setting.general.theme": "Donkere modus",
```
Comment (gnunicorn) [inline src/locales/de.json:174]:
```suggestion
  "setting.general.force_darkmode": "Dunkel-Modus",
```

File: src/locales/en.json [hunk 6]
```diff
@@ -170,6 +170,7 @@
   "richEditor.actions.unorderedListIcon" : "&#8226",
   "richtEditor.defaults.placeholder" : "Text...",
   "setting.general.language": "Language",
+  "setting.general.theme": "Dark Mode",
```
Comment (gnunicorn) [inline src/locales/en.json:173]:
```suggestion
  "setting.general.force_darkmode": "force Dark Mode",
```

File: src/views/settings/General.vue [hunk 7]
```diff
@@ -1,28 +1,33 @@
...
+            </ion-select>
+          </ion-item>
+
+           <ion-item>
+            <ion-label>
+            {{ $t('setting.general.theme') }}
+            </ion-label>
+            <ion-toggle 
+            @ion-change="themeChange(themes = !themes)"
+            :checked="themes" slot="end"/>
```
Comment (gnunicorn) [inline src/views/settings/General.vue:26]:
```suggestion
            @ion-change="forceDark($event.detail.checked)"
            :checked="force_darkmode" slot="end"/>
```

File: src/views/settings/General.vue [hunk 8]
```diff
@@ -58,12 +64,12 @@ export default defineComponent({
...
-      hasPush: computed(() => store.getters["auth/hasPush"]),
-      langChange: (lang: string) => store.dispatch("auth/setLang", lang),
-      SUPPORTED_LANGUAGES,
-      generalIcon,
-      notificationIcon,
-    };
+      themes: computed(() => store.state.auth.user?.settings?.themes),
+      hasPush: computed(() => store.getters['auth/hasPush']),
+      langChange:  (lang: string) => store.dispatch("auth/setLang", lang),
+      themeChange: (themes: boolean) => {store.dispatch("auth/setSetting", {themes}); document.body.classList.toggle('dark', themes);  },
```
Comment (gnunicorn) [inline src/views/settings/General.vue:70]:
```suggestion
      forceDark: (force_darkmode: boolean) => {store.dispatch("auth/setSetting", {force_darkmode}); document.body.classList.toggle('dark', force_darkmode);  },
```

**Review by gnunicorn — APPROVED** [review]
Nice thanks,

but let's also rename the value to have the proper name.

File: src/views/settings/General.vue [hunk 9]
```diff
@@ -1,28 +1,33 @@
...
+              <ion-select-option
+                v-for="opt in SUPPORTED_LANGUAGES"
+                :value="opt[0]"
+                :key="opt[0]">{{opt[1]}}</ion-select-option>
+            </ion-select>
+          </ion-item>
+
+           <ion-item>
+            <ion-label>
+            {{ $t('setting.general.theme') }}
```
Comment (gnunicorn) [inline src/views/settings/General.vue:22]:
```suggestion
            {{ $t('setting.general.force_darkmode') }}
```

File: src/config/App.vue [hunk 10]
```diff
@@ -92,11 +92,12 @@ export default defineComponent({
     if (store.state.auth.wantsToLogin) {
       updateLoginModal(true, false);
     }
-
+        
     watch(
       () => store.state.auth.user,
       async (newVal, oldVal) => {
         if (newVal && newVal != oldVal) {
+          document.body.classList.toggle('dark', newVal?.settings?.force_darkmode);
```
Comment (gnunicorn) [inline src/config/App.vue:100]:
```suggestion
          document.body.classList.toggle('dark', newVal?.settings?.forceDarkMode);
```

File: src/locales/de.json [hunk 11]
```diff
@@ -171,6 +171,7 @@
   "richEditor.actions.unorderedListIcon" : "&#8226",
   "richtEditor.defaults.placeholder" : "Text...",
   "setting.general.language": "Sprache",
+  "setting.general.force_darkmode": "Dunkel-Modus",
```
Comment (gnunicorn) [inline src/locales/de.json:174]:
```suggestion
  "setting.general.forceDarkMode": "Dunkel-Modus",
```

File: src/locales/en.json [hunk 12]
```diff
@@ -170,6 +170,7 @@
   "richEditor.actions.unorderedListIcon" : "&#8226",
   "richtEditor.defaults.placeholder" : "Text...",
   "setting.general.language": "Language",
+  "setting.general.force_darkmode": "force Dark Mode",
```
Comment (gnunicorn) [inline src/locales/en.json:173]:
```suggestion
  "setting.general.forceDarkMode": "force Dark Mode",
```

File: src/views/settings/General.vue [hunk 13]
```diff
@@ -1,28 +1,33 @@
...
+              <ion-select-option
+                v-for="opt in SUPPORTED_LANGUAGES"
+                :value="opt[0]"
+                :key="opt[0]">{{opt[1]}}</ion-select-option>
+            </ion-select>
+          </ion-item>
+
+           <ion-item>
+            <ion-label>
+            {{ $t('setting.general.force_darkmode') }}
```
Comment (gnunicorn) [inline src/views/settings/General.vue:22]:
```suggestion
            {{ $t('setting.general.forceDarkMode') }}
```

File: src/views/settings/General.vue [hunk 14]
```diff
@@ -1,28 +1,33 @@
...
+            </ion-label>
+            <ion-toggle 
+            @ion-change="forceDark($event.detail.checked)"
+            :checked="force_darkmode" slot="end"/>
```
Comment (gnunicorn) [inline src/views/settings/General.vue:26]:
```suggestion
            :checked="forceDarkMode" slot="end"/>
```

File: src/views/settings/General.vue [hunk 15]
```diff
@@ -58,12 +64,12 @@ export default defineComponent({
...
-      hasPush: computed(() => store.getters["auth/hasPush"]),
-      langChange: (lang: string) => store.dispatch("auth/setLang", lang),
-      SUPPORTED_LANGUAGES,
-      generalIcon,
-      notificationIcon,
-    };
+      themes: computed(() => store.state.auth.user?.settings?.themes),
+      hasPush: computed(() => store.getters['auth/hasPush']),
+      langChange:  (lang: string) => store.dispatch("auth/setLang", lang),
+      forceDark: (force_darkmode: boolean) => {store.dispatch("auth/setSetting", {force_darkmode}); document.body.classList.toggle('dark', force_darkmode);  },
```
Comment (gnunicorn) [inline src/views/settings/General.vue:70]:
```suggestion
      forceDark: (forceDarkMode: boolean) => {store.dispatch("auth/setSetting", {forceDarkMode}); document.body.classList.toggle('dark', forceDarkMode);  },
```

**Review by gnunicorn — APPROVED** [review]
style

**Pull request merged**